### PR TITLE
KAFKA-13930: Add 3.2.0 to core upgrade and compatibility system tests

### PR DESCRIPTION
**Pull request by cadonna**
Apache Kafka 3.2.0 was recently released. Now we need
to test upgrades and compatibility with 3.2 in core system tests.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: tests/kafkatest/tests/core/upgrade_test.py [hunk 1]
```diff
@@ -91,6 +94,9 @@ def perform_upgrade(self, from_kafka_version, to_message_format_version=None):
             self.wait_until_rejoin()
 
     @cluster(num_nodes=6)
+    @parametrize(from_kafka_version=str(LATEST_3_2), to_message_format_version=None, compression_types=["none"])
```
Comment (hachikuji) [inline tests/kafkatest/tests/core/upgrade_test.py:97]: Shall we update downgrade_test as well?

File: tests/kafkatest/tests/client/client_compatibility_produce_consume_test.py [hunk 2]
```diff
@@ -73,6 +75,7 @@ def min_cluster_size(self):
     @parametrize(broker_version=str(LATEST_2_8))
     @parametrize(broker_version=str(LATEST_3_0))
     @parametrize(broker_version=str(LATEST_3_1))
+    @parametrize(broker_version=str(LATEST_3_2))
     def test_produce_consume(self, broker_version, metadata_quorum=quorum.zk):
```
Comment (hachikuji) [inline tests/kafkatest/tests/client/client_compatibility_produce_consume_test.py:79]: I wonder if it's useful to continue testing older versions that we have not touched in years. Could we draw the line at 2.4 for example? For older versions, the main thing that seems useful to continue testing is the upgrade/downgrade tests. We don't necessarily have to change this in this patch of course, I'm just curious to hear what people think. cc @ijuma

**Review by hachikuji — COMMENTED** [review]
Thanks @cadonna . Left a couple comments

File: tests/kafkatest/tests/core/upgrade_test.py (hunk 1 above)
Comment (cadonna) [reply tests/kafkatest/tests/core/upgrade_test.py:97]: I included also downgrade tests.

**Review by hachikuji — APPROVED** [review]
LGTM. Thanks @cadonna . Before you merge, it would be helpful if you link to a passing test run.

**Comment by cadonna** [comment]
@hachikuji Thanks for the review! I have already tried to trigger a system test run but there seems to be a environmental issue because I get authentication errors. Locally, the system tests run but they are many and they put quite some load on my laptop even if I filter by the updated tests in this PR. I will retry to run them today and follow up with the responsible folks if I will run into issues.

**Comment by cadonna** [comment]
Passing system test run: http://confluent-kafka-branch-builder-system-test-results.s3-us-west-2.amazonaws.com/2022-06-02--001.system-test-kafka-branch-builder--1654200227--cadonna--AK13930-add_32_upgrade_comp_tests-core--7818cb8df6/report.html

**Comment by cadonna** [comment]
Build failures are unrelated.

File: tests/kafkatest/tests/client/client_compatibility_produce_consume_test.py (hunk 2 above)
Comment (jsancio) [reply tests/kafkatest/tests/client/client_compatibility_produce_consume_test.py:79]: @hachikuji I agree with you. Kafka's system test are getting longer and longer to run. Should we move this discussion the mailing list or Jira to get some agreement here? cc @ijuma

Comment (ijuma) [reply tests/kafkatest/tests/client/client_compatibility_produce_consume_test.py:79]: We also need to test client protocol compatibility with all the versions. I submitted KIP-896 recently to fix this in a holistic way.

**Pull request merged**